File: web/src/models/job/kramerius-job-instance.ts
import { KrameriusJob } from 'enums/kramerius-job'
import { DomainObject } from 'models/domain/domain-object'
import { MapType } from 'models/map-type'
import { JobExecution } from './job-execution'
import { LastLaunch } from './job-launch'

export interface KrameriusJobInstance extends DomainObject {
	executionStatus: string
	executions: JobExecution[]
	jobType: KrameriusJob
	jobParameters: MapType
	lastLaunch: LastLaunch
}

File: api/krameriusplus-core/src/main/java/cz/inqool/dl4dh/krameriusplus/core/job/KrameriusJobInstance.java
package cz.inqool.dl4dh.krameriusplus.core.job;

import cz.inqool.dl4dh.krameriusplus.api.batch.ExecutionStatus;
import cz.inqool.dl4dh.krameriusplus.api.batch.KrameriusJobType;
import cz.inqool.dl4dh.krameriusplus.core.domain.jpa.object.DomainObject;
import cz.inqool.dl4dh.krameriusplus.core.job.report.StepRunReport;
import lombok.Getter;
import lombok.Setter;
import org.springframework.batch.core.JobParameters;

import javax.persistence.*;
import javax.validation.constraints.NotNull;
import java.util.HashMap;
import java.util.Map;

@Getter
@Setter
@Entity
@Table(name = "kplus_kramerius_job_instance")
public class KrameriusJobInstance extends DomainObject {

    @NotNull
    @Enumerated(EnumType.STRING)
    protected ExecutionStatus executionStatus = ExecutionStatus.CREATED;

    protected Long jobInstanceId;

    @OneToMany(mappedBy = "job")
    @MapKey(name = "stepExecutionId")
    private Map<Long, StepRunReport> reports = new HashMap<>();

    @NotNull
    @Enumerated(EnumType.STRING)
    private KrameriusJobType jobType;

    @Convert(converter = JobParametersJsonConverter.class)
    private JobParameters jobParameters;

    @Embedded
    private LastLaunch lastLaunch = new LastLaunch();
}
